keras_nlp/layers/positional_embedding.py

```diff
@@ -0,0 +1,93 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Positional embedding implementation based on `keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class PositionalEmbedding(keras.layers.Layer):
+    """Creates a positional embedding.
+
+    Args:
+      max_length: The maximum size of the dynamic sequence.
+      initializer: The initializer to use for the embedding weights. Defaults to
+        "glorot_uniform".
+      seq_axis: The axis of the input tensor where we add the embeddings.
+
+    Example:
+    ```python
+    positional_embedding = PositionalEmbedding(max_length=100)
+    inputs = keras.Input((100, 32), dtype=tf.float32)
+    outputs = positional_embedding(inputs)
+    ```
+
+    Reference:
+      [BERT: Pre-training of Deep Bidirectional Transformers for Language
+      Understanding](https://arxiv.org/abs/1810.04805).
+    """
+
+    def __init__(
+        self, max_length, initializer="glorot_uniform", seq_axis=1, **kwargs
+    ):
+
+        super().__init__(**kwargs)
+        if max_length is None:
+            raise ValueError("`max_length` must be an Integer, not `None`.")
+        self.max_length = max_length
+        self.initializer = keras.initializers.get(initializer)
+        self.seq_axis = seq_axis
+
+    def get_config(self):
+        config = {
+            "max_length": self.max_length,
+            "initializer": keras.initializers.serialize(self.initializer),
+            "seq_axis": self.seq_axis,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def build(self, input_shape):
+        dimension_list = input_shape.as_list()
+        width = dimension_list[-1]
+        weight_sequence_length = self.max_length
+
+        self.positional_embeddings = self.add_weight(
+            "embeddings",
+            shape=[weight_sequence_length, width],
+            initializer=self.initializer,
+        )
+
+        super().build(input_shape)
+
+    def call(self, inputs):
+        """Forward pass of the PositionalEmbedding.
+
+        Args:
+            inputs: a Tensor. The length along the `seq_axis` is the length of
+            the sequence.
+
+        Returns:
+            A Tensor of the same shape as the `inputs` containing the positional
+            embedding weights for the input sequence.
+        """
+        input_shape = tf.shape(inputs)
+        actual_seq_len = input_shape[self.seq_axis]
+        positional_embeddings = self.positional_embeddings[:actual_seq_len, :]
```

Comment:
I am a bit confused about this part: I think it is to broadcast the positional_embedding along the batch axis, why do we need to reset the other 2 dim (sequence_length and embedding_dim)?

Reply:
You're right, it seems like the intermediary `tf.reshape` is useless, I'll simplify the code